### PR TITLE
expand: Stop using artificial `ast::Item` for macros loaded from metadata

### PR DESCRIPTION
**Pull request by petrochenkov**
You don't need a full `Item` for that, and not using a piece of AST helps with https://github.com/rust-lang/rust/pull/131808.

### CODE REVIEW

**Review by compiler-errors — APPROVED** [review]

**Comment by compiler-errors** [comment]
very nice, r=me when green

**Comment by compiler-errors** [comment]
gotta fix tools it seems, those should be trivial i think. r=me or pass back over to me if you think it needs a re-review.

@rustbot author

File: src/librustdoc/html/format.rs [hunk 1]
```diff
@@ -554,10 +554,8 @@ fn generate_macro_def_id_path(
     // Check to see if it is a macro 2.0 or built-in macro.
     // More information in <https://rust-lang.github.io/rfcs/1584-macros.html>.
     let is_macro_2 = match cstore.load_macro_untracked(def_id, tcx) {
-        LoadedMacro::MacroDef(def, _) => {
-            // If `ast_def.macro_rules` is `true`, then it's not a macro 2.0.
-            matches!(&def.kind, ast::ItemKind::MacroDef(ast_def) if !ast_def.macro_rules)
-        }
+        // If `ast_def.macro_rules` is `true`, then it's not a macro 2.0.
+        LoadedMacro::MacroDef { def, .. } => !def.macro_rules,
```
Comment (klensy) [inline src/librustdoc/html/format.rs:558]: Comment need fix too: `ast_def`

**Comment by compiler-errors** [comment]
Thanks! @bors r+ rollup

**Pull request merged**